src/main/resources/org/jenkinsci/plugins/workflow/libs/SCMBasedRetriever/help-clone.html

```diff
@@ -0,0 +1,8 @@
+<div>
+    If checked, every build performs a fresh clone of the SCM rather than locking and updating a common copy.
+    No changelog will be computed.
+    For Git, you are advised to add <b>Advanced clone behaviors</b>
+    and then check <b>Shallow clone</b> and <b>Honor refspec on initial clone</b> and uncheck <b>Fetch tags</b>
+    to make the clone much faster.
+    You may still enable <b>Cache fetched versions on controller for quick retrieval</b> if you prefer.
+</div>
```

Comment:
Not sure which “library folder” you are referring to here. Can you give an example (path name)?

By default (cache not enabled, clone mode not enabled), the Git repo is checked out into some `workspace` dir and updated there. The `libs` directory inside a build only ever contained `src`, `vars`, and `resources`.

If you enable clone mode, the `libs` directory inside a build is similar, there is just no `workspace` dir.

Not sure offhand what the behaviors were with a cache enabled, but there should be no user-visible change. In all cases, before or after this PR, the `libs` directory inside a build has just the files required for use as a library.

Reply:
Possibly you were referring to https://github.com/jenkinsci/pipeline-groovy-lib-plugin/blob/4a8971b1ad8b758a74561a057b78428ee695ae54/src/main/java/org/jenkinsci/plugins/workflow/libs/SCMBasedRetriever.java#L186 which is just informational: the Git repo is cloned to `/var/jenkins_home/jobs/xxx/builds/123/libs/deadbeef1234/` and then files not needed for Pipeline are immediately removed. This is not really any different from the non-clone-mode behavior where the Git repo is cloned to `/var/jenkins_home/workspace/xxx@libs/` (and left intact there) and then the files which _are_ needed for Pipeline are copied to `/var/jenkins_home/jobs/xxx/builds/123/libs/deadbeef1234/`.

Maybe I should just delete the message to avoid confusion?

Reply:
I found the message helpful to indicate that something was being removed from the repository, but if that is not a change from the earlier end result, then I think it is worth deleting the message so that "message hawks" like me are not distracted by them.

Reply:
We can close this suggestion as not sensible, since it would be documenting behavior that is already there, without any benefit to the user.

Reply:
I removed the messages from the build log.